src/proc/rotation-filter.cpp

```diff
@@ -220,6 +279,38 @@ namespace librealsense {
             }
         }
     }
+
+
+    void rotation_filter::rotate_YUYV_frame(uint8_t* const out, const uint8_t* source, int width, int height) {
+        int frame_size = width * height * 2;  // YUYV has 2 bytes per pixel
+        for( int i = 0; i < frame_size; i += 4 )
+        {
+            // Find the corresponding flipped position in the output
+            int flipped_index = frame_size - 4 - i;
+
+            // Copy YUYV pixels in reverse order to rotate 180 degrees
+            out[flipped_index] = source[i];          // Y1
+            out[flipped_index + 1] = source[i + 1];  // U
+            out[flipped_index + 2] = source[i + 2];  // Y2
+            out[flipped_index + 3] = source[i + 3];  // V
+        }
+    }
+
+
+    bool rotation_filter::should_process( const rs2::frame & frame )
+    {
+        if( ! frame || frame.is< rs2::frameset >() )
+            return false;
+        auto type = frame.get_profile().stream_type();
+        /// change extrinsics
+        for( auto & stream_type : _streams_to_rotate ){
```

Comment:
Consider using `std::find` instead of looping yourself. Might be better optimized.
[For instance](https://stackoverflow.com/questions/75124694/why-does-using-stdfind-generates-different-and-slower-assembly-code-than-ha)